pkg/nimble/netif/include/nimble_netif.h

```diff
@@ -0,0 +1,229 @@
+ * corresponding GNRC thread.
+ *
+ * Although the wrapper hooks into GNRC using the netif interface, it does need
+ * to implement parts of the netdev interface as well. This is done where
+ * needed.
+ *
+ * # Usage
+ * This submodule is designed to work fully asynchronous, in the same way as the
+ * NimBLE interfaces are designed. All functions in this submodule will only
+ * trigger the intended action. Once this action is complete, the module will
+ * report the result asynchronously using the configured callback.
+ *
+ * So before using this module, make sure to register a callback using the
+ * nimble_netif_eventcb() function.
+ *
+ * After this, this module provides functions for managing BLE connections to
+ * other devices. Once these connections are established, this module takes care
+ * of mapping IP packets to the corresponding connections.
+ *
+ * @{
+ *
+ * @file
+ * @brief       GNRC netif implementation for NimBLE
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NIMBLE_NETIF_H
+#define NIMBLE_NETIF_H
+
+#include <stdint.h>
+
+#include "net/ble.h"
+#include "net/ipv6.h"
+
+#include "host/ble_hs.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Default L2CAP channel ID to use
+ */
+#ifndef NIMBLE_NETIF_CID
+#define NIMBLE_NETIF_CID            (BLE_L2CAP_CID_IPSP)
+#endif
+
+/**
+ * @brief   Default MTU size supported by the NimBLE netif wrapper
+ */
+#ifndef NIMBLE_NETIF_MTU
+#define NIMBLE_NETIF_MTU            (IPV6_MIN_MTU)
+#endif
+
+/**
+ * @brief   Return codes used by the NimBLE netif module
+ */
+enum {
+    NIMBLE_NETIF_OK         =  0,   /**< everything went fine */
+    NIMBLE_NETIF_NOTCONN    = -1,   /**< not connected */
+    NIMBLE_NETIF_DEVERR     = -2,   /**< internal BLE stack error */
+    NIMBLE_NETIF_BUSY       = -3,   /**< network device is busy */
+    NIMBLE_NETIF_NOMEM      = -4,   /**< insufficient memory */
+    NIMBLE_NETIF_NOTADV     = -5,   /**< not advertising */
+    NIMBLE_NETIF_NOTFOUND   = -6,   /**< no fitting entry found */
+};
+
+/**
+ * @brief   Event types triggered by the NimBLE netif module
+ */
+typedef enum {
+    NIMBLE_NETIF_CONNECTED_MASTER,  /**< connection established as master */
+    NIMBLE_NETIF_CONNECTED_SLAVE,   /**< connection established as slave */
+    NIMBLE_NETIF_CLOSED_MASTER,     /**< connection closed (we were master) */
+    NIMBLE_NETIF_CLOSED_SLAVE,      /**< connection closed (we were slave) */
+    NIMBLE_NETIF_CONNECT_ABORT,     /**< connection establishment aborted */
+    NIMBLE_NETIF_CONN_UPDATED,      /**< connection parameter update done */
+} nimble_netif_event_t;
+
+/**
+ * @brief   Flags describing the state of a single connection context
+ */
+enum {
+    NIMBLE_NETIF_L2CAP_CLIENT       = 0x0001,   /**< l2cap client */
+    NIMBLE_NETIF_L2CAP_SERVER       = 0x0002,   /**< l2cap server */
+    NIMBLE_NETIF_L2CAP_CONNECTED    = 0x0003,   /**< l2cap is connected */
```

Comment:
Consistency: Sometimes you write `l2cap` sometimes `L2CAP`? I think the capitalized form should be used in writing.